homeassistant/components/flux_led/light.py

```diff
@@ -139,41 +147,66 @@
 )
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the platform and manage importing from YAML."""
+
+    if config.get("automatic_add", False):
+        await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data={CONF_TYPE: "auto"}
+        )
+
+    else:
+        for import_host, import_item in config.items():
+            if import_host == "platform":
+                continue
+
+            import_name = import_host
+            if import_item:
+                import_name = import_item.get("name", import_host)
+
+            await hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data={
+                    CONF_TYPE: "manual",
+                    CONF_NAME: import_name,
+                    CONF_HOST: import_host,
+                },
+            )
+
+    return True
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
     """Set up the Flux lights."""
+    config_data = hass.data[DOMAIN][entry.entry_id]
+    config_type = config_data[CONF_TYPE]
     lights = []
-    light_ips = []
 
-    for ipaddr, device_config in config.get(CONF_DEVICES, {}).items():
+    if config_type == "auto":
+        # Find the bulbs on the LAN
+        scanner = BulbScanner()
+        scanner.scan(timeout=10)
```

Comment:
I'm assuming this isn't async safe.

Reply:
This is not async safe.

Reply:
Ok. Then it needs to run in the executor with `hass.async_add_executor_job`.

Reply:
Addressed.